### PR TITLE
[#19] [#20] [UI] [Integrate] As a user, I can see Survey Detail Questions Screen

### LINKED ISSUE #19: [UI] As a user, I can see Survey Detail Questions Screen
## Why

* User need to see survey title and description

## Acceptance Criteria

* Show survey title, description, and start survey button

## Design

Screen Shot 2022-05-10 at 09 00 34

### LINKED ISSUE #20: [Integration] As a signed-in user, I can view survey detail
## Why

* Allow user to navigate to the survey detail screen

## Acceptance Criteria

* Navigate the user to the survey detail screen when the user taps the circle arrow button
* Show survey detail screen with actual data

### PR DESCRIPTION
**Pull request by ducbm051291**
- Close #19 
- Close #20 

## What happened 👀

- Create SurveyDetailsScreen, SurveyDetailsState, SurveyDetailsViewModel
- Update AppRoute and RoutePath for SurveyDetailScreen
- Create SurveyDetailsComponentId, SurveyDetailsSkeletonLoading
- Rename HomeLoading to LoadingText
- Add convert function in SurveyModel to convert to SurveyDetailsUIModel
- Write UnitTest for SurveyDetailsViewModel

## Insight 📝

- N/A

## Proof Of Work 📹

CleanShot 2023-05-08 at 16 18 16

### CODE REVIEW

File: lib/ui/survey_details/survey_details_screen.dart [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            loading: () => const SurveyDetailsSkeletonLoading(),
+            success: (uiModel) => _surveyDetails(uiModel),
+          );
+        },
+      );
+
+  Widget _surveyDetails(SurveyDetailsUIModel uiModel) => Consumer(
+        builder: (_, ref, __) {
+          return Container(
+            width: MediaQuery.of(context).size.width,
```
Comment (nmint8m) [inline lib/ui/survey_details/survey_details_screen.dart:76]:
```suggestion
            width: context.screenSize.width,
```

File: lib/ui/survey_details/survey_details_screen.dart [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            padding: const EdgeInsets.all(Dimensions.paddingMedium),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Text(
+                  uiModel.title,
+                  style: Theme.of(context).textTheme.displayLarge,
```
Comment (nmint8m) [inline lib/ui/survey_details/survey_details_screen.dart:84]: Please define `textTheme` as a part of `BuildContextExtension`.

```suggestion
                  style: context.textTheme.displayLarge,
```

File: lib/ui/survey_details/survey_details_screen.dart [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                ),
+                const SizedBox(
+                  height: Dimensions.paddingNormal,
+                ),
```
Comment (nmint8m) [inline lib/ui/survey_details/survey_details_screen.dart:88]: As there is only 1 argument for creating a gap widget, let's keep it simple in 1 line.

```suggestion
                const SizedBox(height: Dimensions.paddingNormal),
```

File: lib/ui/survey_details/survey_details_screen.dart [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        .read(surveyDetailsViewModelProvider.notifier)
+        .getSurveyDetails(widget.surveyId);
+  }
+
+  void _startSurvey() {}
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
```
Comment (nmint8m) [inline lib/ui/survey_details/survey_details_screen.dart:150]: We can remove this code for now.

**Review by nmint8m — COMMENTED** [review]
There a few comments. Please check 👍

File: lib/ui/survey_details/survey_details_screen.dart [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  Widget get _emptyBody => Container();
+
+  Widget get _body => Consumer(
+        builder: (_, ref, __) {
+          final state = ref.watch(surveyDetailsViewModelProvider);
+          return state.maybeWhen(
+            orElse: () => _emptyBody,
+            loading: () => const SurveyDetailsSkeletonLoading(),
+            success: (uiModel) => _surveyDetails(uiModel),
+          );
```
Comment (thiennguyen0196) [inline lib/ui/survey_details/survey_details_screen.dart:69]: Ditto here 
```suggestion
            loading: () => const SurveyDetailsSkeletonLoading(),
            success: (uiModel) => _surveyDetails(uiModel),
            orElse: () => _emptyBody,
          );
```

**Review by thiennguyen0196 — COMMENTED** [review]
Just some minor concerns, rest lgtm

**Review by nmint8m — APPROVED** [review]
Look good to me. 👍

**Pull request merged**